src/scipp/core/assignments.py

```diff
@@ -8,6 +8,21 @@
 from ..typing import DataArray, Dataset, Variable
 
 
+def _combine_args(arg0: Optional[Dict[str, Variable]] = None, /, **kwargs):
```

Comment:
Still duplicate with https://github.com/scipp/scipp/blob/62778d23843b4f5cf1fd7e5e00ddfc201eea721a/src/scipp/core/dimensions.py#L12-L21. You can keep the error and not worry about it (since it should be extremely rare), or adjust it there to only print the keys?